### PR TITLE
feat(ui): implement dark mode

### LINKED ISSUE #116: Implement dark mode
### Context

Debugger currently provides only **light mode**, in order to have a seamless experience for developers we need to add support for **dark mode**. 
Dark Mode will be configurable from app settings: 

Image

### PR DESCRIPTION
**Pull request by kreshnnikgashi**
This closes #116 and #123

### CODE REVIEW

**Comment by danstarns** [comment]
This is looking great, some initial comments, and expect more to come. 

**Npm Icon**
I find this jarring, it seems to be off to one side. 

@kreshnnikgashi I think we should just remove the npm icon now and direct only to the GitHub - wdyt?

Screenshot 2023-11-10 at 12 01 53




**Config Icon**
Using the user icon could be confusing to developers as we also have a user for the login, Let's keep anything user-related to the cloud and keep this as a cog only. 

Screenshot 2023-11-10 at 11 55 49


**Editor Color**
I think the purple in the dark background is hard to see, try something a little bit lighter. 

Screenshot 2023-11-10 at 11 56 41

**Comment by danstarns** [comment]
I see we are having issues with the modal colors, I think we should change the modal background to be the same as the trace views, so that we dont have to invert the pilll or background colors, wdyt? @kreshnnikgashi 

Screenshot 2023-11-10 at 12 04 24

File: apps/ui/src/__mocks__/matchMedia.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Object.defineProperty(window, "matchMedia", {
+  writable: true,
+  value: jest.fn().mockImplementation((query) => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // Deprecated
```
Comment (danstarns) [inline apps/ui/src/__mocks__/matchMedia.ts:7]: Why are we using deprecated, methods, can we change to the latest?

Comment (kreshnnikgashi) [reply apps/ui/src/__mocks__/matchMedia.ts:7]: Actually this is the latest: https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom - however I'm gonna remove these 2 deprecated methods

**Comment by kreshnnikgashi** [comment]
> I see we are having issues with the modal colors, I think we should change the modal background to be the same as the trace views, so that we dont have to invert the pilll or background colors, wdyt? @kreshnnikgashi
> 
> Screenshot 2023-11-10 at 12 04 24

I agree @danstarns

**Comment by kreshnnikgashi** [comment]
> I see we are having issues with the modal colors, I think we should change the modal background to be the same as the trace views, so that we dont have to invert the pilll or background colors, wdyt? @kreshnnikgashi
> 
> Screenshot 2023-11-10 at 12 04 24

I agree @danstarns  - should this be the case for all modals or only for trace viewer ?

**Comment by danstarns** [comment]
> > I see we are having issues with the modal colors, I think we should change the modal background to be the same as the trace views, so that we dont have to invert the pilll or background colors, wdyt? @kreshnnikgashi
> > Screenshot 2023-11-10 at 12 04 24
> 
> I agree @danstarns - should this be the case for all modals or just for trace viewer ?

For all modals!

**Review by danstarns — APPROVED** [review]
It looks like it's the flaky color e2e tests blocking this. Let's skip those lines and make a separate issue linking to each line.

Great work @kreshnnikgashi  let's merge and 🚢 it.

**Pull request merged**